tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -710,40 +710,20 @@ class Restrictions {
     var classDefinition = documentDefinition;
     if (classDefinition is! ClassDefinition) return errors;
 
-    var fieldType = classDefinition.findField(parentNodeName)?.type;
-
-    if (fieldType == null) return errors;
-
-    errors.addAll(_validateFieldDataType(fieldType, span));
-
     var field = classDefinition.findField(parentNodeName);
-    if (field == null || !field.isSymbolicRelation) return errors;
+    if (field == null) return errors;
 
-    if (!type.endsWith('?')) {
-      errors.add(SourceSpanSeverityException(
-        'Fields with a model relations must be nullable (e.g. $parentNodeName: $type?).',
-        span,
-      ));
-    }
+    errors.addAll(_validateFieldDataType(field.type, span));
 
+    // Abort further validation if the field data type has errors.
     if (errors.isNotEmpty) return errors;
 
-    String? parsedType = parsedModels.extractReferenceClassName(field);
-
-    var referenceClass = parsedModels.findByClassName(parsedType);
-
-    if (referenceClass is! ClassDefinition) {
-      errors.add(SourceSpanSeverityException(
-        'Only classes can be used in relations, "$parsedType" is not a class.',
-        span,
-      ));
-      return errors;
-    }
-
-    if (!_hasTableDefined(referenceClass)) {
-      errors.add(SourceSpanSeverityException(
-        'The class "$parsedType" must have a "table" property defined to be used in a relation.',
-        span,
+    if (field.isSymbolicRelation) {
```

Comment:
Question: Are we able to refactor this `isSymbolicRelation` I feel like this is a left over concept that probably shouldn't be here. But I don't exactly remember the implementation details.

Reply:
What kind of a refactor are you looking for? Renaming or a different approach?

This is the definition of the field:
```dart
  /// Returns true, if this field has a relation pointer to/from another field
  /// with relation type [ForeignRelationDefinition]. This means that this field
  /// relation does not propagate to the database, but instead is managed by
  /// the other field.
  bool get isSymbolicRelation =>
    relation != null && relation is! ForeignRelationDefinition;
  ```

The description, together with the implementation, makes me confused as well. So, I'm not super confident in making any change here at this moment without building more knowledge around it.

Reply:
Yeah this might be a little too annoying to deal with right now. So the variable means this is a relation field but is not stored in the db. I think it also has to be an int? I.E. no object or list relations. 

I don't remember why that is significant or if it could be refactored into another approach. But lets skip deal with that another day.